File: src/easy/Q0009.cpp
// 9. Palindrome Number
// https://leetcode.com/problems/palindrome-number/

#include <string>

using namespace std;

// Time: O(n), Space: O(1), where n is the number of digits in `x`.
bool is_palindrome(int x) {
  auto x_str = to_string(x);
  size_t left = 0, right = x_str.size() - 1;
  while (left < right) {
    if (x_str[left] != x_str[right]) {
      return false;
    }
    ++left;
    --right;
  }
  return true;
}

File: src/easy/Q0009/Q0009.py
"""
9. Palindrome Number
https://leetcode.com/problems/palindrome-number/
"""


# Time: O(n), Space(n), where n is the number of digits in `x`.
def is_palindrome(x: int) -> bool:
    s = str(x)
    lo, hi = 0, len(s) - 1
    while lo < hi:
        if s[lo] != s[hi]:
            return False
        lo += 1
        hi -= 1
    return True
